eth/abc.py

```diff
@@ -334,10 +335,53 @@ def is_genesis(self) -> bool:
         ...
 
 
+class WitnessIndexAPI(ABC):
+    @property
+    @abstractmethod
+    def hashes(self) -> Tuple[Hash32, ...]:
+        ...
+
+    @property
+    @abstractmethod
+    def accounts_queried(self) -> Collection[Address]:
+        ...
+
+    @property
+    @abstractmethod
+    def account_bytecodes_queried(self) -> Tuple[Address, ...]:
+        ...
+
+    @abstractmethod
+    def get_slots_queried(self, address: Address) -> Tuple[int, ...]:
+        ...
+
+    @property
+    @abstractmethod
+    def total_slots_queried(self) -> int:
+        """
+        Summed across all accounts, how many storage slots were queried?
+        """
+        ...
+
+
+class BlockEvalResult(NamedTuple):
```

Comment:
Something like `BlockAndWitnessSketch` would make me less likely to want to look up the definition, I only know what a `BlockEvalResult` is once I've looked it up.